pyk/src/tests/integration/proof/test_mini_kevm.py

```diff
@@ -106,3 +168,87 @@ def test_all_path_reachability_prove(
 
             assert proof.status == proof_status
             assert leaf_number(proof) == expected_leaf_number
+
+    @pytest.mark.parametrize(
+        'test_id,spec_file,spec_module,claim_id,max_iterations,max_depth,cut_rules,proof_status,expected_leaf_number',
+        CUSTOM_STEP_TEST_DATA,
+        ids=[test_id for test_id, *_ in CUSTOM_STEP_TEST_DATA],
+    )
+    def test_custom_step(
+        self,
+        kprove: KProve,
+        kcfg_explore: KCFGExplore,
+        test_id: str,
+        spec_file: str,
+        spec_module: str,
+        claim_id: str,
+        max_iterations: int | None,
+        max_depth: int | None,
+        cut_rules: Iterable[str],
+        proof_status: ProofStatus,
+        expected_leaf_number: int,
+        tmp_path_factory: TempPathFactory,
+    ) -> None:
+        global skip_custom_step
+
+        skip_custom_step = False
```

Comment:
This looks problematic. What's the purpose of this variable?

Reply:
The idea of the test is to run a proof twice, once with a custom_step, and once with the rule from the K semantics, and assert that the kcfgs are the same. The global var is used to skip the custom step for the second execution of the proof.

Reply:
But then there's a dependency between test runs. Also, this only works if you run the two tests on the same runner.

Why not just write a very simple K definition, and add two tests: one with a custom step, one without, and assert the end result for both.

Reply:
I don't see the dependency. Both proofs are executed in the same `test_custom_proof` defined below. The alternative I see would be to define two implementations of a KCFGSemantics: one with a custom_step implementation, and one without and to use both in the same test to execute the same claim and assert the results.

Reply:
> I don't see the dependency

IIUC one test run sets the global variable and another one reads it.

> The alternative I see would be to define two implementations of a KCFGSemantics: one with a custom_step implementation, and one without and to use both in the same test to execute the same claim and assert the results.

That's what I was thinking. You can make this a parameterized test where the semantics is an input parameter.

Reply:
> IIUC one test run sets the global variable and another one reads it.

No, the `skip_custom_step` is used only in this test, `test_custom_step`. There are two proofs `proof1` and `proof2` of the same claim, with two init_cterms and two provers, one for each proof. I'm running the first proof with `skip_custom_step` set to `False`, to execute the custom_step, then I'm setting the `skip_custom_step` to `True` and I execute the second proof.

Reply:
I see, so it's a way to configure the behavior of a single `KCFGSemantics` instance. Still, the need for a global variable indicates the existing set of fixtures is not a good fit for the test as currently implemented.

Solutions I can think of:
- Implement a custom fixture that lets you create `KCFGExplore` instances with different `KCFGSemantics` instances plugged in during test execution.
- The one we already discussed: https://github.com/runtimeverification/k/pull/4376#discussion_r1616222653

Also, please write a simple, dedicated semantics for this test.